app/lib/homework/homework_dialog/homework_dialog.dart

```diff
@@ -362,11 +362,19 @@ class _SaveButton extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return SaveButton(
-      key: HwDialogKeys.saveButton,
-      tooltip: "Hausaufgabe speichern",
-      onPressed: () => onPressed(context),
-    );
+    bool isSaving = false;
+    return StatefulBuilder(builder: (context, setState) {
+      return SaveButton(
+        key: HwDialogKeys.saveButton,
+        tooltip: "Hausaufgabe speichern",
+        onPressed: isSaving
+            ? null
+            : () {
+                setState(() => isSaving = true);
+                onPressed(context);
```

Comment:
I'm pretty sure this causes a bug if an error occurs during the upload (e.g. network issue) because then the save button is still disabled and the user can't retry.

I think the following change could fix it:

```dart
setState(() => isSaving = true);
await onPressed(context);
setState(() => isSaving = false);
```